server/report.cpp

```diff
@@ -1062,8 +1062,25 @@ static void dem_line_item(char *outptr, size_t out_size,
                           struct player *pplayer, const demographic &demo,
                           bv_cols selcols)
 {
+  int value = -1;
+  if (!pplayer || BV_ISSET(selcols, DEM_COL_QUANTITY)
+      || BV_ISSET(selcols, DEM_COL_RANK)
+      || BV_ISSET(selcols, DEM_COL_BEST)) {
+    if (pplayer->score.demographics.count(demo.name()) > 0) {
```

Comment:
Codacy wants to say that global observers requesting demographics crash the server.

Reply:
For reference, one gets more context with `clang-tidy -p compile_commands.json server/report.cpp --checks=clang-analyzer-core.NullDereference`.